code/game/objects/machinery/cryopod.dm

```diff
@@ -39,37 +119,96 @@
+			state = HS_CNSL_OTHER
+	updateUsrDialog()
+
 	if(href_list["item"])
-		var/obj/item/I = locate(href_list["item"]) in GLOB.cryoed_item_list
+		var/obj/item/I 
+		switch(state)
+			if(HS_CNSL_GUN)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_gun
+			if(HS_CNSL_AMMO)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_ammo
+			if(HS_CNSL_EXPLOSIVE)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_explosive
+			if(HS_CNSL_MELEE)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_melee
+			if(HS_CNSL_CLOTHING)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_clothing
+			if(HS_CNSL_FOOD)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_food
+			if(HS_CNSL_DRUGS)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_drugs
+			if(HS_CNSL_CONTAINERS)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_containers
+			if(HS_CNSL_OTHER)
+				I = locate(href_list["item"]) in GLOB.cryoed_item_list_other
 		dispense_item(I, usr)
 
 	else if(href_list["allitems"])
 
-		if(!length(GLOB.cryoed_item_list))
+		if(!length(GLOB.cryoed_item_list_gun) && !length(GLOB.cryoed_item_list_ammo) && !length(GLOB.cryoed_item_list_explosive) && !length(cryoed_item_list_melee) && !length(GLOB.cryoed_item_list_clothing) && !length(GLOB.cryoed_item_list_food) && !length(GLOB.cryoed_item_list_drugs) && !length(GLOB.cryoed_item_list_containers) && !length(GLOB.cryoed_item_list_other))
 			to_chat(usr, span_warning("There is nothing to recover from storage."))
 			updateUsrDialog()
 			return
 
 		visible_message(span_notice("[src] beeps happily as it disgorges the desired objects."))
 
-		for(var/obj/item/I AS in GLOB.cryoed_item_list)
+		for(var/obj/item/I in GLOB.cryoed_item_list_gun || GLOB.cryoed_item_list_ammo || GLOB.cryoed_item_list_explosive || GLOB.cryoed_item_list_melee || GLOB.cryoed_item_list_clothing || GLOB.cryoed_item_list_food || GLOB.cryoed_item_list_drugs || GLOB.cryoed_item_list_containers || GLOB.cryoed_item_list_other)
 			dispense_item(I, usr, FALSE)
 
 	updateUsrDialog()
 
 
 /obj/machinery/computer/cryopod/proc/dispense_item(obj/item/I, mob/user, message = TRUE)
 	if(!istype(I) || QDELETED(I))
-		GLOB.cryoed_item_list -= I
+		GLOB.cryoed_item_list_gun -= I;
+		GLOB.cryoed_item_list_ammo -= I;
+		GLOB.cryoed_item_list_explosive -= I;
+		GLOB.cryoed_item_list_melee -= I;
+		GLOB.cryoed_item_list_clothing -= I;
+		GLOB.cryoed_item_list_food -= I;
+		GLOB.cryoed_item_list_drugs -= I;
+		GLOB.cryoed_item_list_containers -= I;
+		GLOB.cryoed_item_list_other -= I;
 		CRASH("Deleted or erroneous variable ([I]) called for hypersleep inventory retrivial.")
-	if(!(I in GLOB.cryoed_item_list))
+	if(!I in (GLOB.cryoed_item_list_gun || GLOB.cryoed_item_list_ammo || GLOB.cryoed_item_list_explosive || GLOB.cryoed_item_list_melee || GLOB.cryoed_item_list_clothing || GLOB.cryoed_item_list_food || GLOB.cryoed_item_list_drugs || GLOB.cryoed_item_list_containers || GLOB.cryoed_item_list_other))
```

Comment:
You can;t check lists like this.
You have to do !(var in list) x your amount of lists

Reply:
![image](https://github.com/tgstation/TerraGov-Marine-Corps/assets/24830358/fe3f19f1-c786-42ea-adf0-97bd14aa0c1a)
I went for solution B. Didn't notice the nonsensical code.